searx/utils.py

```diff
@@ -831,3 +830,27 @@ def js_variable_to_python(js_variable):
     s = s.replace(chr(1), ':')
     # load the JSON and return the result
     return json.loads(s)
+
+
+def parse_timediff_string(timediff_str: str) -> timedelta | None:
+    """Parse a time string in format MM:SS or HH:MM:SS and convert it to a `timedelta` object.
+
+    Returns None if the provided string doesn't match any of the formats.
+    """
```

Comment:
Suggestion to simplify "time_parts":

```python
>>> duration = "12:42"
>>> hour, min, sec = [int(x) for x in (duration.split(":") + ["00"])[:3]]

>>> timedelta(hours=hour, minutes=min, seconds=sec)
datetime.timedelta(seconds=45720)
``

Reply:
The real value would be `datetime.timedelta(seconds=762)`, `["00"]` has to be appended to the start of the list, not the end :)

Reply:
Seems like a good suggestion though :+1: